components/admin_console/team_channel_settings/channel/details/channel_details.tsx

```diff
@@ -421,11 +440,77 @@ export default class ChannelDetails extends React.Component<ChannelDetailsProps,
             privacyChanging = false;
         }
 
-        this.setState({serverError, saving: false, saveNeeded, isPrivacyChanging: privacyChanging});
+        const usersToAddList = Object.values(usersToAdd);
+        const usersToRemoveList = Object.values(usersToRemove);
+        const userRolesToUpdate = Object.keys(rolesToUpdate);
+        const usersToUpdate = usersToAddList.length > 0 || usersToRemoveList.length > 0 || userRolesToUpdate.length > 0;
```

Comment:
I'm not sure about this name(`userRolesToUpdate`), doesn't jive with the other two imo, but I don't have a better solution (maybe `usersWithRolesToUpdate`) 1/5 to look for a better name.